### PR TITLE
Extract and send metainfo to Sematext backend

### PR DESCRIPTION
**Pull request by bsmid**
This PR covers full metainfo logic:
- extraction from Telegraf metrics data
- serialization into format understood by Sematext backend
- sending the metainfo in a separate goroutine with retry logic in case response failed because of 5xx server error (exponential backoff, max 10 tries)

### CODE REVIEW

File: plugins/outputs/sematext/processors/metainfo.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+	return metrics, nil
+}
+
+func (m *Metainfo) sendMetainfo(newMetrics map[string]*MetricMetainfo) {
+	reqCounter := 0
+
+	mInfoSlice := make([]*MetricMetainfo, len(newMetrics))
+	for _, mInfo := range newMetrics {
+		mInfoSlice = append(mInfoSlice, mInfo)
```
Comment (rabbitstack) [inline plugins/outputs/sematext/processors/metainfo.go:120]: if initial slice capacity is equal to `len(newMetrics)`, wouldn't `append` start adding the elements after `len(newMetrics)` nil entries?

File: plugins/outputs/sematext/processors/metainfo.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+			nextSleepIntervalSec := int32(reqCounter * metainfoRetryIntervalSeconds)
+			m.log.Infof("metainfo sending retry in %d seconds", nextSleepIntervalSec)
+			time.Sleep(time.Second * time.Duration(rand.Int31n(nextSleepIntervalSec)))
+		}
+	}
+}
+
+func processMetric(token string, metric *telegraf.Metric, field *telegraf.Field,
+	sentMetrics *map[string]*MetricMetainfo) (*MetricMetainfo, string) {
```
Comment (rabbitstack) [inline plugins/outputs/sematext/processors/metainfo.go]: why are you passing a pointer to map?

File: plugins/outputs/sematext/processors/metainfo.go [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+	host, set := (*metric).GetTag(telegrafHostTag)
```
Comment (rabbitstack) [inline plugins/outputs/sematext/processors/metainfo.go]: (*metric).GetTag(telegrafHostTag) -> metric.GetTag(telegrafHosTag) ?

File: plugins/outputs/sematext/processors/metainfo.go [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+	if numericType == UnsupportedNumericType || numericType == Bool {
+		return nil
+	}
+
+	label := fmt.Sprintf("%s.%s", (*metric).Name(), field.Key)
+
+	return &MetricMetainfo{
+		token:       token,
+		name:        field.Key,
+		namespace:   (*metric).Name(),
```
Comment (rabbitstack) [inline plugins/outputs/sematext/processors/metainfo.go]: metric.Name() ?

File: plugins/outputs/sematext/processors/metainfo.go (hunk 1 above)
Comment (bsmid) [reply plugins/outputs/sematext/processors/metainfo.go:120]: Yes, a missing 0 in make.

**Review by rabbitstack — APPROVED** [review]

File: plugins/outputs/sematext/processors/metainfo.go (hunk 2 above)
Comment (bsmid) [reply plugins/outputs/sematext/processors/metainfo.go]: Pass by reference, but maps already are pointers... will fix.

File: plugins/outputs/sematext/processors/metainfo.go (hunk 3 above)
Comment (bsmid) [reply plugins/outputs/sematext/processors/metainfo.go]: Was passing an interface by reference, fixed it now.

**Pull request merged**